File: Sample/NendSDK_Sample/FeedView.m
//
//  FeedView.m
//  NendSDK_Sample
//
//  Copyright © 2015年 F@N Communications. All rights reserved.
//

#import "FeedView.h"

@interface FeedView ()

@property (nonatomic, weak) IBOutlet UILabel *title;
@property (nonatomic, weak) IBOutlet UILabel *category;
@property (nonatomic, weak) IBOutlet UIActivityIndicatorView *indicator;

@end

@implementation FeedView

- (void)awakeFromNib
{
    [super awakeFromNib];
    
    UITapGestureRecognizer *recognizer = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(tapped:)];
    [self addGestureRecognizer:recognizer];
    
    if (@available(iOS 13.0, *)) {
        self.backgroundColor = [UIColor colorNamed:@"NativeAdBackgroundColor"];
        self.title.textColor = [UIColor colorNamed:@"TextColor"];
        self.category.textColor = [UIColor colorNamed:@"TextColor"];
    }
}

- (void)setTitleText:(NSString *)titleText
{
    self.title.text = titleText;
}

- (void)setCategoryText:(NSString *)categoryText
{
    self.category.text = categoryText;
}

- (void)setAdLoading:(BOOL)adLoading
{
    if (!self.indicator) {
        return;
    }
    if (adLoading) {
        self.title.text = @"";
        self.category.text = @"";
        [self.indicator startAnimating];
        self.indicator.hidden = NO;
    } else {
        [self.indicator stopAnimating];
        self.indicator.hidden = YES;
    }
}

- (void)tapped:(id)sender
{
    [[UIApplication sharedApplication] openURL:[NSURL URLWithString:self.link]];
}


@end

File: Sample/NendSDK_Swift_Sample/FeedView.swift
//
//  FeedView.swift
//  NendSDK_Sample
//
//  Copyright © 2015年 F@N Communications. All rights reserved.
//

import UIKit

class FeedView: UIView {
    
    @IBOutlet fileprivate weak var title: UILabel!
    @IBOutlet fileprivate weak var category: UILabel!
    @IBOutlet fileprivate weak var indicator: UIActivityIndicatorView?
    
    override func awakeFromNib() {
        super.awakeFromNib()
        
        let recognizer = UITapGestureRecognizer(target: self, action: #selector(FeedView.tapped(_:)))
        self.addGestureRecognizer(recognizer)
    }
    
    var titleText: String? {
        didSet {
            self.title.text = self.titleText
        }
    }
    
    var categoryText: String? {
        didSet {
            self.category.text = self.categoryText
        }
    }
    
    var adLoading: Bool = false {
        didSet {
            if let indicator = self.indicator {
                if self.adLoading {
                    self.title.text = "";
                    self.category.text = "";
                    indicator.startAnimating();
                    indicator.isHidden = false;
                } else {
                    indicator.stopAnimating()
                    indicator.isHidden = true
                }
            }
        }
    }
    
    var link: String?
    
    func tapped(_ sender: AnyObject) {
        UIApplication.shared.openURL(URL(string: self.link!)!)
    }
}
